File: abelkhan_rpc_demo/proto/client_call_hub/js/module/ccallhmodule.js
/*this rsp file is codegen by abelkhan for js*/

function rsp_ccallh(_hub, _uuid)
{
    this.hub_handle = _hub;
    this.uuid = _uuid;
    this.call = function(argv0)
    {
        _hub.gates.call_client(_hub.gates.current_client_uuid, "ccallh", "ccallh_rsp", _uuid, argv0);
    }
    this.err = function()
    {
        _hub.gates.call_client(_hub.gates.current_client_uuid, "ccallh", "ccallh_err", _uuid);
    }
}
module.exports.rsp_ccallh = rsp_ccallh;

function ccallh(_hub)
{
    var event_cb = require("event_cb");
    event_cb.event_cb.call(this);

    this.module_name = "ccallh";
    this.hub_handle = _hub;
    _hub.modules.add_module("ccallh", this);

    this.ccallh = function(uuid)
    {
        _hub.modules.rsp = new rsp_ccallh(_hub, uuid);
        this.call_event("ccallh", []);
        _hub.modules.rsp = null;
    }

}
module.exports.ccallh = ccallh;

File: abelkhan_rpc_demo/proto/client_call_hub/csharp/module/ccallhmodule.cs
/*this rsp file is codegen by abelkhan for c#*/

using System;
using System.Collections;
using System.Collections.Generic;

using abelkhan;

namespace rsp
{
    public class rsp_ccallh : abelkhan.Response
    {
        public string uuid;

        public rsp_ccallh(string _uuid)
        {
            uuid = _uuid;
        }

        public void call(String argv0)
        {
            hub.hub.gates.call_client(hub.hub.gates.current_client_uuid, "ccallh", "ccallh_rsp", uuid, argv0);
        }
        public void err()
        {
            hub.hub.gates.call_client(hub.hub.gates.current_client_uuid, "ccallh", "ccallh_err", uuid);
        }
    }

    public class ccallh_module : abelkhan.Module
    {
        public string module_name;
        public hub.hub hub_handle;
        public ccallh_module(hub.hub _hub)
        {
            module_name = "ccallh";
            hub_handle = _hub;
            hub.hub.modules.add_module("ccallh", this);
        }

        public delegate void ccallhhandle();
        public event ccallhhandle onccallh;
        public void ccallh(string uuid)
        {
            if(onccallh != null)
            {
                rsp = new rsp_ccallh(uuid);
                onccallh();
                rsp = null;
            }
        }

    }
}
